packages/desktop/components/drawers/network-config/drawers/components/AddIscpChainForm.svelte

```diff
@@ -0,0 +1,107 @@
+<script lang="ts">
+    import { localize } from '@core/i18n'
+    import { IIscpChainMetadata, MAX_CHAIN_NAME_LENGTH } from '@core/network'
+    import { ChainType } from '@core/network/enums'
+    import { isValidHexAddress, isValidUrl } from '@core/utils'
+    import { Button, HTMLButtonType, Input } from '@ui'
+
+    const localeKey = 'views.dashboard.drawers.networkConfig.addChain'
+
+    const isBusy = false
+    let nameError = ''
+    let aliasAddressError = ''
+    let iscpEndpointError = ''
+    let explorerUrlError = ''
+    $: submitDisabled = !chain.name || !chain.aliasAddress || !chain.iscpEndpoint
+
+    const chain: IIscpChainMetadata = {
+        type: ChainType.Iscp,
+        chainId: undefined,
+        name: '',
+        explorerUrl: undefined,
+        aliasAddress: '',
+        iscpEndpoint: '',
+    }
+
+    function validateName(): void {
+        if (!chain.name) {
+            nameError = localize(`${localeKey}.errors.cannotBeEmpty`)
+        } else if (chain.name.length > MAX_CHAIN_NAME_LENGTH) {
+            nameError = localize(`${localeKey}.errors.nameTooLong`)
+        }
+    }
+
+    function validateAliasAddress(): void {
+        if (!isValidHexAddress(chain.aliasAddress)) {
+            aliasAddressError = localize(`${localeKey}.errors.aliasAddressMustBeHex`)
+        }
+    }
+
+    function validateIscpEndpoint(): void {
+        if (!isValidUrl(chain.iscpEndpoint)) {
+            iscpEndpointError = localize(`${localeKey}.errors.invalidUrl`)
+        }
+    }
+
+    function validateExplorerUrl(): void {
+        if (chain.explorerUrl && !isValidUrl(chain.explorerUrl)) {
+            explorerUrlError = localize(`${localeKey}.errors.invalidUrl`)
+        }
+    }
+
+    function validate(): void {
+        validateName()
+        validateAliasAddress()
+        validateIscpEndpoint()
+        validateExplorerUrl()
```

Comment:
I think part of the validation should include if either the name or alias address is already being used. The explorer URL is fine and the ISCP endpoint is fine so long as the alias address is different.

Reply:
Might be difficult to test that logic but the logic itself should be straightforward, might change slightly once we add the classes in :sweat_smile: